mirgecom/inviscid.py

```diff
@@ -77,29 +77,20 @@ def inviscid_flux(discr, eos, q):
 
 
 def get_inviscid_timestep(discr, eos, cfl, q):
-    """Routine (will) return the (local) maximum stable inviscid timestep.
-
-    Currently, it's a hack waiting for the geometric_factor helpers port
-    from grudge.
-    """
+    """Routine returns the cell-local maximum stable inviscid timestep."""
     dim = discr.dim
-    mesh = discr.mesh
-    order = max([grp.order for grp in discr.discr_from_dd("vol").groups])
-    nelements = mesh.nelements
-    nel_1d = nelements ** (1.0 / (1.0 * dim))
+    cv = split_conserved(dim, q)
 
-    # This roughly reproduces the timestep AK used in wave toy
-    dt = (1.0 - 0.25 * (dim - 1)) / (nel_1d * order ** 2)
-    return cfl * dt
+    from grudge.dt_utils import (dt_non_geometric_factor,
+                                 dt_geometric_factors)
+    from mirgecom.fluid import compute_wavespeed
 
-#    dt_ngf = dt_non_geometric_factor(discr.mesh)
-#    dt_gf  = dt_geometric_factor(discr.mesh)
-#    wavespeeds = compute_wavespeed(w,eos=eos)
-#    max_v = clmath.max(wavespeeds)
-#    return c*dt_ngf*dt_gf/max_v
+    return (
+        cfl * dt_non_geometric_factor(discr) * dt_geometric_factors(discr)
+        / compute_wavespeed(dim, eos, cv)
+    )
 
 
 def get_inviscid_cfl(discr, eos, dt, q):
     """Calculate and return CFL based on current state and timestep."""
```

Comment:
Maybe clarify in the doc that this returns a node-wise CFL approximation?

Reply:
see how this grabs ya. (80d8087)